src/test/basic.test.ts

```diff
@@ -1040,4 +1040,260 @@ test(
+
+      /**
+       * The script that we expect to be cancelled.
+       *
+       * It's important that this script does not depend on the one that fails,
+       * because that would be the easy case.
+       */
+      const cancel = await rig.newCommand();
+
+      /**
+       * Gives us control over when the `cancel` script starts.
+       */
+      const cancelBlocker = await rig.newCommand();
+
+      /**
+       * Gives us control over when the `failParent` script resolves.
+       */
+      const failParentBlocker = await rig.newCommand();
+
+      await rig.write({
+        'package.json': {
+          scripts: {
+            main: 'wireit',
+            failParent: 'wireit',
+            failParentBlocker: 'wireit',
+            fail: 'wireit',
+            cancel: 'wireit',
+            cancelBlocker: 'wireit',
+          },
+          wireit: {
+            main: {
+              dependencies: ['failParent', 'cancel'],
+            },
+            failParent: {
+              dependencies: ['fail', 'failParentBlocker'],
+            },
+            failParentBlocker: {
+              command: failParentBlocker.command,
+            },
+            fail: {
+              command: fail.command,
+            },
+            cancel: {
+              command: cancel.command,
+              dependencies: ['cancelBlocker'],
+            },
+            cancelBlocker: {
+              command: cancelBlocker.command,
+            },
+          },
+        },
+      });
+
+      const wireit = rig.exec('npm run main', {
+        env: {
+          WIREIT_FAILURES: envSetting,
+        },
+      });
+
+      // `fail`, `failParentBlocker`, and `cancelBlocker` start
+      const failInv = await fail.nextInvocation();
+      const failParentBlockerInv = await failParentBlocker.nextInvocation();
+      const cancelBlockerInv = await cancelBlocker.nextInvocation();
+
+      // The failure occurs.
+      failInv.exit(1);
+
+      // Unblock `cancel`. It could start, but it shouldn't, because a failure has
+      // occured elsewhere in the graph. Wait a moment first to ensure Wireit
+      // notices the failure before we unblock.
+      await new Promise((resolve) => setTimeout(resolve, 50));
+      cancelBlockerInv.exit(0);
+
+      // Unblock `failParent`. We do this after unblocking `cancel` so that we
+      // cover the case where the branch that failed has not yet reached the root
+      // of the graph, because it's easy to imagine an implementation that relies
+      // on that. Wait a moment first to ensure Wireit unblocks `cancel` before
+      // `fail`.
+      await new Promise((resolve) => setTimeout(resolve, 50));
+      failParentBlockerInv.exit(0);
+
+      assert.equal((await wireit.exit).code, 1);
+      assert.equal(fail.numInvocations, 1);
+      assert.equal(failParentBlocker.numInvocations, 1);
+      assert.equal(cancel.numInvocations, 0);
+      assert.equal(cancelBlocker.numInvocations, 1);
```

Comment:
It's not exactly the target of the test but should this also include `assert.equal(failParent.numInvocations, 0);` ?

Reply:
failParent doesn't have a `command`, so there's nothing to verify for it